### PR TITLE
refactor: switch to C++11-compatible tr_wait() from tr_wait_msec()

### PR DESCRIPTION
**Pull request by dmantipov**

### CODE REVIEW

File: libtransmission/utils.h [hunk 1]
```diff
@@ -15,6 +15,10 @@
 #include <string_view>
 #include <type_traits>
 #include <vector>
+#include <thread>
+#include <chrono>
```
Comment (ckerr) [inline libtransmission/utils.h:19]: _nit_ please sort these two additions in with the other std headers in lines 8-17

**Review by ckerr — COMMENTED** [review]
LGTM, one fewer `#ifdef`

File: libtransmission/utils.h (hunk 1 above)
Comment (dmantipov) [reply libtransmission/utils.h:19]: OK

**Review by ckerr — APPROVED** [review]
Thanks!

File: libtransmission/utils.h [hunk 2]
```diff
@@ -7,15 +7,19 @@
...
 #include <cstddef> // size_t
 #include <ctime> // time_t
 #include <optional>
 #include <string>
 #include <string_view>
+#include <thread>
 #include <type_traits>
 #include <vector>
 
+using namespace std::chrono_literals;
```
Comment (nevack) [inline libtransmission/utils.h:21]: Is it good practice?
Feels like `using namespace ...` in header file pollutes consumer namespace.

Comment (dmantipov) [reply libtransmission/utils.h:21]: Well,  consistent usage of  `std::chrono::duration` literals through the whole code base might be a notable exception.

Comment (ckerr) [reply libtransmission/utils.h:21]: @nevack is write; this goes against [this cpp core guideline](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#sf7-dont-write-using-namespace-at-global-scope-in-a-header-file) and will generate a new sonarcloud warning when merged.  Let's try removing this from the header and see how cumbersome it gets; we can always add it back in later if the cc-level cleanup is a burden.

**Review by ckerr — CHANGES_REQUESTED** [review]
@nevack's suggestion of removing `using namespace std::chrono_literals;` from header

File: libtransmission/utils.h (hunk 2 above)
Comment (dmantipov) [reply libtransmission/utils.h:21]: OK

File: libtransmission/utils.h [hunk 3]
```diff
@@ -85,8 +87,12 @@ constexpr auto tr_saveFile(std::string_view filename, ContiguousRange const& x,
 /** @brief return the current date in milliseconds */
 [[nodiscard]] uint64_t tr_time_msec();
 
-/** @brief sleep the specified number of milliseconds */
-void tr_wait_msec(long int delay_milliseconds);
+/** @brief sleep for the specified duration */
+template<class rep, class period>
+void tr_wait(std::chrono::duration<rep, period> const& delay)
```
Comment (nevack) [inline libtransmission/utils.h:92]: nit: Maybe move definition to `utils.cc`?

**Review by nevack — APPROVED** [review]
LGTM

File: libtransmission/utils.h (hunk 3 above)
Comment (ckerr) [reply libtransmission/utils.h:92]: A smaller, no-new-dependencies template function like this makes more sense in the header IMO

**Review by ckerr — APPROVED** [review]

**Pull request merged**